python/datashaper/datashaper/engine/verbs/verbs_mapping.py

```diff
@@ -31,6 +43,98 @@ def inner(func: Callable[..., TableContainer]) -> Callable[..., TableContainer]:
     return inner
 
 
+def new_row(old_tuple: tuple, new_column_name: str, value: Any) -> tuple:
+    """Helper function to return rows in row-wise operations."""
+    old_named_tuple_type = type(old_tuple)
+    return namedtuple(
+        old_named_tuple_type.__name__, old_named_tuple_type._fields + (new_column_name,)
+    )(*(old_tuple + (value,)))
+
+
+def parallel_verb(
+    name: str,
+    treats_input_tables_as_immutable: bool = False,
+    override_existing: bool = False,
+    asyncio_type: str = "asyncio",
+    **kwargs,
+) -> Callable:
+    """Decorator for registering a parallel verb."""
+
+    def inner(func: Callable[..., pd.DataFrame]) -> Callable[..., TableContainer]:
+        @verb(
+            name=name,
+            treats_input_tables_as_immutable=treats_input_tables_as_immutable,
+            override_existing=override_existing,
+        )
+        async def wrapper(
+            input: VerbInput,
+            callbacks: Any,
+            max_parallelism: int = 4,
+            *args,
+            **kwargs,
+        ):
+            input_table = input.source.table
+            chunk_size = kwargs.pop("chunk_size", 1)
+            if chunk_size == 1:
+                chunks = input_table.itertuples()
+            elif (
+                chunk_size > 0
+                and signature(func).parameters["chunk"].annotation == pd.DataFrame

```

Comment:
With datashaper we sometimes use our `Table` type alias - do you think this annotation mechanism is safe?

Reply:
I changed it to Table... tbh I hate that type alias, this is just used because of the groupby verb in pandas returns a different type to be aggregated later, I'd like to see if we could either change the way this work or deprecate that verb since we already have the aggregate which does a groupby and aggregate back to a dataframe type.